dask_planner/src/parser.rs

```diff
@@ -0,0 +1,221 @@
+                    Keyword::DROP => {
+                        // move one token forward
+                        self.parser.next_token();
+                        // use custom parsing
+                        self.parse_drop()
+                    }
+                    _ => {
+                        // use the native parser
+                        Ok(DaskStatement::Statement(Box::from(
+                            self.parser.parse_statement()?,
+                        )))
+                    }
+                }
+            }
+            _ => {
+                // use the native parser
+                Ok(DaskStatement::Statement(Box::from(
+                    self.parser.parse_statement()?,
+                )))
+            }
+        }
+    }
+
+    /// Parse a SQL CREATE statement
+    pub fn parse_create(&mut self) -> Result<DaskStatement, ParserError> {
+        match self.parser.parse_identifier() {
+            Ok(ident) => match ident.value.to_lowercase().as_str() {
+                "create" => self.parse_create_model(),
+                _ => Ok(DaskStatement::Statement(Box::from(
+                    self.parser.parse_create()?,
+                ))),
+            },
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Parse a SQL DROP statement
+    pub fn parse_drop(&mut self) -> Result<DaskStatement, ParserError> {
+        match self.parser.parse_identifier() {
+            Ok(ident) => match ident.value.to_lowercase().as_str() {
+                "model" => self.parse_drop_model(),
+                _ => Ok(DaskStatement::Statement(Box::from(
+                    self.parser.parse_drop()?,
+                ))),
+            },
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Parse Dask-SQL CREATE MODEL statement
+    fn parse_create_model(&mut self) -> Result<DaskStatement, ParserError> {
+        let model_name = self.parser.parse_object_name()?;
+        self.parser.expect_keyword(Keyword::WITH)?;
+        self.parser.expect_token(&Token::LParen)?;
+
+        // Parse all KV pairs into a Vec<BinaryExpr> instances
+        let kv_binexprs = self.parser.parse_comma_separated(Parser::parse_expr)?;
+
+        let _kv_pairs: HashMap<String, &Box<Expr>> = kv_binexprs
+            .iter()
+            .map(|f| match f {
+                Expr::BinaryOp { left, op: _, right } => (
+                    match *left.clone() {
+                        Expr::Value(value) => match value {
+                            Value::EscapedStringLiteral(key_val)
+                            | Value::SingleQuotedString(key_val)
+                            | Value::DoubleQuotedString(key_val) => key_val,
+                            _ => "".to_string(),
+                        },
+                        _ => "".to_string(),
+                    },
+                    right,
+                ),
+                _ => panic!("Expected BinaryOp, Key/Value pairs, found: {}", f),
+            })
+            .collect();
```

Comment:
We can avoid the panic here and use Result instead.

```suggestion
        let _kv_pairs: Vec<(String, &Box<Expr>)> = kv_binexprs
            .iter()
            .map(|f| match f {
                Expr::BinaryOp { left, op: _, right } => match *left.clone() {
                    Expr::Value(value) => match value {
                        Value::EscapedStringLiteral(key_val)
                        | Value::SingleQuotedString(key_val)
                        | Value::DoubleQuotedString(key_val) => Ok((key_val, right)),
                        _ => Ok(("".to_string(), right)),
                    },
                    _ => Ok(("".to_string(), right)),
                },

                _ => parser_err!(format!("Expected BinaryOp, Key/Value pairs, found: {}", f)),
            })
            .collect::<Result<Vec<_>, ParserError>>()?;
        let _kv_pairs: HashMap<String, &Box<Expr>> = _kv_pairs.into_iter().collect();

```

Reply:
The magic here is that returning results in the `iter` gives us a `Vec<Result<_>>` which we can transform into `<Result<Vec_>>` using the turbofish notation in `collect::<Result<Vec<_>, ParserError>>()?`